### DISCUSSION TITLE
Mainline NPU kernel driver

### DISCUSSION THREAD

**Discussion by Ferenc-** [Joshua-Riek/ubuntu-rockchip#876] [Ideas]
Is [this development](https://blog.tomeuvizoso.net/2024/06/rockchip-npu-update-4-kernel-driver-for.html) something that can be considered for the experimental mainline kernel?

**Comment by Joshua-Riek** [comment]
I do not have any experimental mainline Linux images for users. While I work on mainline sometimes it's to much of a hassle to maintain, especially since RK3588 is still so poorly supported.

**Comment by bagong** [comment]
6.11 or 12 should be better...

**Reply by Joshua-Riek** [reply]
I would like to release some Ubuntu 24.10 images using mainline Linux, as the non LTS Ubuntu releases are only supported for 9 months anyway.

**Reply by StuartIanNaylor** [reply]
That would be great as there are a few things in 6.10 with panthor, the npu which IMO sucks as really its a 2tops x3 and that is when using a small ram partition to hold weight files.
The mainline NPU driver is there so interesting to test though and also if the NUMA additions also helps with the rk3588.
I haven't seen anything schedules for 6,11 or 6.12 that is not in 6.10 though so 6.10 would actually be great for just an experimental image.
